### PR TITLE
Menu: add arrows for subMenus and update alignments in Material

### PR DESCRIPTION
**Pull request by AlexanderMoiseev**
All changes are made for Material:
- added arrow for subMenus
- item is moved to flex
- logical properties are used for RTL
- fixed text alignment
- icon is standartised
- margins and paddings changed according to design
- grid's menu border fixed

### CODE REVIEW

File: packages/devextreme/scss/widgets/material/menu/_index.scss [hunk 1]
```diff
@@ -24,22 +25,53 @@ $material-menu-font-size: $material-base-font-size;
...
 
     &.dx-menu-item-has-text {
+      .dx-menu-item-text {
+        flex: 1 1 auto;
+      }
+
       &.dx-menu-item-has-icon {
         .dx-icon {
-          margin-right: $material-menu-horizontal-padding;
+          margin-right: $material-menu-icon-to-text-offset;
```
Comment (EugeniyKiyashko) [inline packages/devextreme/scss/widgets/material/menu/_index.scss]: Let's use logical property here - margin-inline-end

File: packages/devextreme/scss/widgets/material/menu/_index.scss [hunk 2]
```diff
@@ -50,7 +82,7 @@ $material-menu-font-size: $material-base-font-size;
       &.dx-menu-item-has-text {
         &.dx-menu-item-has-icon {
           .dx-icon {
-            margin-left: $material-menu-item-with-icon-horizontal-padding - $material-menu-horizontal-padding - $material-base-icon-size;
+            margin-left: $material-menu-item-with-icon-horizontal-padding - $material-menu-icon-to-text-offset - $material-button-icon-size;
```
Comment (EugeniyKiyashko) [inline packages/devextreme/scss/widgets/material/menu/_index.scss]: After using logical property you may remove this block of styles

File: packages/devextreme/scss/widgets/base/_menu.scss [hunk 3]
```diff
@@ -12,6 +12,7 @@
 
   .dx-menu-item-wrapper {
     display: inline-block;
+    vertical-align: middle;
```
Comment (marker-dao) [inline packages/devextreme/scss/widgets/base/_menu.scss:15]: Can we use flex-box here?

**Review by pomahtri — DISMISSED** [review]

**Review by EugeniyKiyashko — DISMISSED** [review]

File: packages/devextreme/scss/widgets/base/_menu.scss (hunk 3 above)
Comment (AlexanderMoiseev) [reply packages/devextreme/scss/widgets/base/_menu.scss:15]: yes, we can set display flex to dx-menu-items-container, but in this case we should apply flex-direction depending on whether menu orientation is vertical or horizontal

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by marker-dao — APPROVED** [review]

**Review by williamvinogradov — APPROVED** [review]

**Pull request merged**